### ISSUE TITLE
Thank you for the extension and all of the detailed information at :7851

### ISSUE THREAD

**Issue by RandomInternetPreson** [erew123/alltalk_tts#1]
Really nice implementation!!  I like the detailed deepspeed information you supplied in the readme and settings!! I normally do windows installations, and I really like all the documentation you supplied, really great thank you!

**Comment by erew123** [comment]
Thanks! I really appreciate it! And thanks for trying it out, as far as I know, you're the first to use it.

**Comment by erew123** [comment]
@RandomInternetPreson fyi.. I did just post 1x minor update to deal with a gradio issue in the dropdowns. If you want you can just download **server.py** and drop it over the top of your current one.

**Comment by erew123** [comment]
@RandomInternetPreson and I also updated how the narrator splits text between voice/narrator (hopefully to improve it) as well as populate the settings voices with the voices in the folder. I also added a couple more voices.

So you might want to git pull it again or just drop the server.py, tty_server.py and \templates\generate_form.html over the top of the others!

Sorry, still early days and I'm just improving a few little bits while I remember.

**Comment by RandomInternetPreson** [comment]
You are good! I almost got all of the deepspeed stuff working for windows, I actually managed to get through all the steps but I think I needed to take more care when doing the compiling.  Next on the list is to try deepspeed with a WSL install, but I'm pretty happy with the extension even without deepspeed working for now.

Thanks again, going to do a new text-gen-install to take advantage of all the new updates oobabooga's code, and your extension in on my extension installation list now <3

**Comment by erew123** [comment]
One final update done just now. So if you had updated before, its worth an update again. I've done a few more bits that should keep the generated audio a bit more on track/keeping it closer to the original.

And I really do think this is everything done for a full v1. So unless someone says there's bugs, I cant see me changing the code for a good long time (famous last words and clearly tempting fate).

**Comment by erew123** [comment]
Oh and one more thing if you're interested. I couldn't include all these voices within github https://www.file.io/YeR7/download/3RR3Nzx6acnO

Some are better than others....I think there's about 45 there.

**Comment by erew123** [comment]
@RandomInternetPreson changed the link for the voice files... you have 6x days to grab these https://filebin.net/t97nd69ac7qm2rsf if you want them

**Comment by erew123** [comment]
Just fyi. Ive done a major rebuild of the narration engine and filtering engine. The result is a much cleaner output for voice and more control over the narration/character function.

**Comment by erew123** [comment]
FYI, someone found a way to get DeepSpeed v11.2 in the default text-generation-webui environment https://github.com/erew123/alltalk_tts/blob/main/README.md#deepspeed-112-for-windows--python-311

**Comment by RandomInternetPreson** [comment]
Frick this is what I needed, going to try this out today!!

**Comment by RandomInternetPreson** [comment]
FRICK Dude!!! This extension is amazing now, deepspeed on windows working as per your instructions. TY! I've been wrapped up in things and didn't notice your messages :C Are your voices hosted anywhere else?  :3 Thank you again for all the hard work you have put into this!

**Comment by RandomInternetPreson** [comment]
I found where you linked to your voices, thanks again for this extension.  Amazing!

**Comment by erew123** [comment]
Thanks! :) 

Finetuning next along with the new API suite..... just tidying up the documentation for this one!

![image](https://github.com/erew123/alltalk_tts/assets/35898566/a3f29219-3070-43ce-9c83-1e8a60e93482)

**Comment by erew123** [comment]
and, its done https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-finetuning-a-model

**Comment by RandomInternetPreson** [comment]
This is the best tts extension out there!  The level of detail in your instructions is 🔥 next on my list is to fine-tune!  I know others have been asking about this since coqi dropped.  I hope oobabooga makes this one of the default extensions.  Simply the best!

**Comment by erew123** [comment]
Sadly he wont... He said the code base was too complicated and he was worried he would have to do the upkeep of it. I didn't know that was the way it worked aka, he would feel that he would have to take over the code base.

Thanks for the feedback though :) I appreciate it!

**Comment by erew123** [comment]
P.S. yeah... documentation... I hate doing it haha! Its just taken me 7 hours to do all the documentation correctly for the Finetuning...double check it.. reinstall everything from scratch to make sure it works correctly etc etc.....

So thanks for appreciating that aspect of it :)

**Comment by RandomInternetPreson** [comment]
Interesting, I guess that makes sense.  I didn't think about oobabooga needing to maintain the code too, welp I've got your code working well with my current version of booga at least.

Really great work, yeah especially on the documentation it looks like it took a lot of work to put together.  Yeah, the reinstall the double checking everything, the documentation, seems to take longer than writing the code ... :c thanks for doing all of this!!

It's now possible to essentially have real time conversations with the AI in a convenient way, this helps me create training data, I edit the conversations as part of my workflow.  

I don't know if you use the whisper stt extension, but I have an edit to the ui_chat.py file that puts the record button up next to the generate button so you don't need to scroll down to use it:

https://github.com/RandomInternetPreson/MiscFiles/blob/main/WhisperSTTButtonRelocater/recordbutton.py

The combination of your extension and the button makes communicating with the AI faster than typing a lot of the time.

**Comment by erew123** [comment]
Ive not tried the whisper extension, but as the finetuning uses it.. I probably should give it a go. Thanks for the button! :)
